src/ripple/rpc/handlers/LedgerEntry.cpp

```diff
@@ -231,6 +232,22 @@ doLedgerEntry(RPC::JsonContext& context)
             uNodeIndex = keylet::emittedTxn(uNodeIndex).key;
         }
     }
+    else if (context.params.isMember(jss::import_vlseq))
+    {
+        expectedType = ltIMPORT_VLSEQ;
+        if (!context.params[jss::public_key].isString())
+        {
+            uNodeIndex = beast::zero;
+            jvResult[jss::error] = "malformedRequest";
+        }
+        else
+        {
+            auto const pkHex =
+                strUnHex(context.params[jss::public_key].asString());
+            auto const pk = PublicKey(makeSlice(*pkHex));
```

Comment:
I believe the publickey constructor will crash the application if the public key isn't valid, so this is an attack vector.

the pk validity needs to be checked prior to construction